### PR TITLE
Feature/stripe integration

### PR DESCRIPTION
**Pull request by valdrinkoshi**

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
 <ion-nav-view></ion-nav-view>
 
+<script type="text/javascript" src="https://js.stripe.com/v2/"></script>
```
Comment (valdrinkoshi) [inline app/index.html]: i couldn't find another way to include this. I wanted to have it as a bower dependency but i couldn't find it

File: app/scripts/payment.controller.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    .module('HomeCooked.controllers')
+    .controller('PaymentCtrl', PaymentCtrl);
+
+  PaymentCtrl.$inject = ['$scope', '$state', '$ionicPopup', '$ionicLoading', 'HCMessaging', 'PaymentService'];
+
+  function PaymentCtrl($scope, $state, $ionicPopup, $ionicLoading, HCMessaging, PaymentService) {
+
+    var vm = this;
+
+    $scope.stripeCallback = stripeCallback;
```
Comment (valdrinkoshi) [inline app/scripts/payment.controller.js]: need to put this on the scope, as angular-payments directive doesn't accept methods from other scopes...

File: app/templates/settings/payment.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+      <label class="item item-input">
+        <input type="text" ng-model="vm.cvc" placeholder="CVC" payments-format="cvc" payments-validate="cvc"
+               name="cvc" ng-minlength="1"/>
+        <span class="invalid-entry" ng-show="paymentForm.cvc.$dirty && paymentForm.cvc.$invalid">Invalid CVC</span>
+      </label>
+
+      <div class="item button-bar">
+        <button type="submit" class="button button-positive" ng-disabled="paymentForm.$invalid"
+                ng-click="number=vm.number; expiry=vm.expiry; cvc=vm.cvc; showLoading()">
```
Comment (valdrinkoshi) [inline app/templates/settings/payment.html]: ugly, but angular-payments needs this in the scope of the directive :|

**Comment by valdrinkoshi** [comment]
@dbaq can you review please?
You'll need to do grunt install
You might also need to create a stripe account and update the STRIPE_KEY in Gruntfile.js
Also check my comments, there is some ugly stuff that i couldn't avoid, if you know a better solution please let me know

**Comment by valdrinkoshi** [comment]
https://github.com/hmecookd/HomeCooked/issues/24
image

File: app/index.html (hunk 1 above)
Comment (dbaq) [inline app/index.html]: if the app starts offline or with poor connectivity we will have an issue I believe.

**Comment by valdrinkoshi** [comment]
Tests are failing because Stripe.js is not available for tests. I'll search a way to include this as dependency, worst case i manually copy this js file...

File: app/scripts/payment.controller.js (hunk 2 above)
Comment (dbaq) [inline app/scripts/payment.controller.js]: ok, that's interesting and could be reported to their repo.

**Comment by dbaq** [comment]
Looks good to me. :+1:  except one or two comments.

**Comment by valdrinkoshi** [comment]
I'm not so happy with this angular-payments, I'll try something else. It is just handling the form, and calling a Stripe method. Maybe I can do it myself.
About Stripe.js if you find a way to include it through bower instead of having to load it from their server that would be great. Worst case I will manually download that js and include it between our bundled resources...but ideally it would be great to have it integrated through bower!

File: package.json [hunk 4]
```diff
@@ -3,13 +3,14 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "lodash": "^3.5.0"
+    "stripe": "^3.5.0"
```
Comment (valdrinkoshi) [inline package.json]: I currently load it but i don't use it in the app...this should be done through browserify (separate PR)

**Comment by valdrinkoshi** [comment]
@dbaq FYI
Feedback incorporated, merging this PR

**Pull request merged**